core/store/src/trie/shard_tries.rs

```diff
@@ -479,4 +555,66 @@ impl KeyForStateChanges {
             }
         })
     }
+
+    /// Iterates over pairs of `(row key, deserialized row value)` where row key matches `self`.
+    pub fn find_rows_iter<'a>(
+        &'a self,
+        store: &'a Store,
+    ) -> impl Iterator<Item = Result<(Box<[u8]>, RawStateChangesWithTrieKey), std::io::Error>> + 'a
+    {
+        let prefix_len = Self::estimate_prefix_len();
+        debug_assert!(self.0.len() >= prefix_len);
```

Comment:
in such case - let's also print the key (otherwise it would be very hard to debug/analyze)

Reply:
Done